src/main/java/seedu/lifeasier/commands/EditDeadlineCommand.java

```diff
@@ -76,7 +87,8 @@ public void execute(Ui ui, NoteList notes, TaskList tasks, FileStorage storage,
             logger.log(Level.SEVERE, "Input Deadline name does not match any of the existing Deadline names.");
             ui.showNoMatchesMessage("deadline");
         }
-        storage.saveTasks();
+        storage.saveTasks(); //Edit confirmed, can push old saved into tasks arraylist
```

Comment:
Help move the storage.saveTasks() from after all the catch blocks to after when the taskHistory.pushOldCopy() method is called - (Line 74 in this case) for the event and lesson edit classes also.